File: front/src/api/asset.ts
/*
 * 资产归属与可见性
 *
 * 租户资产（知识库 / Skill / IM 绑定 / 模型端点）是「作者所有」的：
 * - private：仅作者与租户管理员（admin/owner）可见；
 * - shared：作者主动共享后，租户内所有人可读，但仍只有作者与管理员可改/删。
 *
 * 后端按 JWT claims + 行内 created_by/visibility 双重校验，前端只负责把
 * 对应按钮显隐到一致，不作为安全边界。
 */

export type AssetVisibility = 'private' | 'shared'

/** 作者才可改/删，管理员可管全部；旧数据 created_by 为空时仅管理员可管。 */
export function canManageAsset(
  row: { created_by?: string } | null | undefined,
  userId: string | undefined,
  managesTenant: boolean,
): boolean {
  if (managesTenant) return true
  return !!row?.created_by && row.created_by === userId
}

/** 共享后的资产对租户只读可见。 */
export function isShared(row: { visibility?: string } | null | undefined): boolean {
  return row?.visibility === 'shared'
}

export function visibilityLabel(v: string | undefined): string {
  return v === 'shared' ? '共享' : '私有'
}

File: trpcservice/domain/asset/asset.go
// Package asset holds the ownership vocabulary shared by every tenant asset
// (knowledge base, skill, channel binding, model endpoint).
//
// An asset records who created it and whether its author has shared it with the
// tenant. The author owns the row; tenant managers (admin/owner) manage every
// row of their tenant. Package asset deliberately owns only the vocabulary —
// who may do what is decided by the web layer from the authenticated claims,
// because that rule also depends on the caller's tenant and role.
package asset

// Visibility values stored in each asset row.
const (
	// VisibilityPrivate means the row is visible to its author and to tenant
	// managers only. This is the default: creating an asset does not publish it.
	VisibilityPrivate = "private"
	// VisibilityShared means the author has published the row to the tenant:
	// every member may read it, but only the author (or a tenant manager) may
	// change or delete it.
	VisibilityShared = "shared"
)

// VisibilityOrDefault normalizes a stored/incoming value, defaulting to private.
// An unrecognized value is treated as private rather than as shared: a typo must
// never widen access.
func VisibilityOrDefault(v string) string {
	if v == VisibilityShared {
		return VisibilityShared
	}
	return VisibilityPrivate
}

// Shared reports whether the visibility makes the row tenant-readable.
func Shared(v string) bool { return v == VisibilityShared }
